app/javascript/controllers/header_controller.js

```diff
@@ -0,0 +1,20 @@
+import { Controller } from "@hotwired/stimulus"
+
+// Connects to data-controller="header"
+export default class extends Controller {
+  static targets = [ "arrow", "circle"]
+  static classes = [ "rotate", "hide"]
+
+  transform_add() {
+    this.arrowTarget.classList.add(this.rotateClass)
+  }
+
+  transform_remove() {
+    this.arrowTarget.classList.remove(this.rotateClass)
+  }
+
+  hide() {
+    this.circleTarget.classList.add(this.hideClass)
+  }
```

Comment:
Let's move this to a different controller. One can be the "drop down controller" and other can be the "notification controller"